lib/did_you_mean/extra_features/key_error.rb

```diff
@@ -0,0 +1,35 @@
+module DidYouMean
+  module Correctable
+    prepend_features KeyError
+  end
+end
+
+module DidYouMean
+  module ExtraFeatures
+    module KeyErrorReciver
+      def fetch(name, *args, &block)
+        super
+      rescue KeyError => e
+        key_names = keys
+        e.define_singleton_method(:keys) { key_names.map(&:inspect) }
+        e.define_singleton_method(:name) { name.inspect }
+        raise e
+      end
+    end
+    Hash.prepend KeyErrorReciver
+    ENV.singleton_class.prepend KeyErrorReciver
+    # TODO: Kernel.sprintf, String#%
+
+    class KeyErrorChecker
+      include SpellCheckable
+      def initialize(key_error)
+        @key_error = key_error
```

Comment:
The key error should be [abstracted away](https://github.com/yuki24/did_you_mean/blob/c4f0247/lib/did_you_mean/spell_checkers/method_name_checker.rb#L6-L10) since [a correctable error creates a spell checker object](https://github.com/yuki24/did_you_mean/blob/00238ef/lib/did_you_mean/core_ext/name_error.rb#L24) and the code above will cause a circular dependency, which may cause memory leaks. I guess MRI handles them pretty well, but I'm not sure about others.